core/domain/suggestion_services.py

```diff
@@ -1826,3 +2020,866 @@ def update_question_suggestion(
+        suggestion_models.QuestionReviewStatsModel.get_all_by_user_id(
+            user_id
+        )
+    )
+    return [
+        _create_question_review_stats_from_model(model)
+        for model in question_review_stats_models
+    ]
+
+
+# TODO(https://shorturl.at/deLSV): pre-fetching and caching of stats data
+# should be done.
+def get_all_contributor_stats(
+    user_id: str
+) -> suggestion_registry.ContributorStatsSummary:
+    """Gets ContributorStatsSummary corresponding to the supplied user.
+
+    Args:
+        user_id: str. User ID.
+
+    Returns:
+        ContributorStatsSummary. ContributorStatsSummary domain objects
+        corresponding to the supplied user.
+    """
+    translation_contribution_stats = get_all_translation_contribution_stats(
+        user_id)
+    translation_review_stats = _get_all_translation_review_stats(user_id)
+    question_contribution_stats = _get_all_question_contribution_stats(user_id)
+    question_review_stats = _get_all_question_review_stats(user_id)
+
+    return suggestion_registry.ContributorStatsSummary(
+        user_id,
+        translation_contribution_stats,
+        question_contribution_stats,
+        translation_review_stats,
+        question_review_stats)
+
+
+def _update_translation_contribution_stats_models(
+    translation_contribution_stats: List[
+        suggestion_registry.TranslationContributionStats
+    ]
+) -> None:
+    """Updates TranslationContributionStatsModel models for given translation
+    contribution stats.
+
+    Args:
+        translation_contribution_stats: list(TranslationContributionStats).
+            A list of TranslationContributionStats domain objects.
+
+    Raises:
+        Exception. Language is None.
+        Exception. Contributor user ID is None.
+        Exception. Topic ID is None.
+    """
+    stats_dict = {}
+    for stat in translation_contribution_stats:
+        if stat.language_code is None:
+            raise Exception('Language code should not be None.')
+        if stat.contributor_user_id is None:
+            raise Exception('Contributor user ID should not be None.')
+        if stat.topic_id is None:
+            raise Exception('Topic ID should not be None.')
+        stat_id = (
+            suggestion_models.TranslationContributionStatsModel.construct_id(
+                str(stat.language_code),
+                str(stat.contributor_user_id),
+                str(stat.topic_id))
+        )
+        stats_dict[stat_id] = stat
+
+    stats_ids = stats_dict.keys()
+
+    stats_models = get_translation_contribution_stats_models(list(stats_ids))
+    stats_models_to_update: List[
+        suggestion_models.TranslationContributionStatsModel] = []
+    for stats_model in stats_models:
+        # We can confirm that stats_model will not be None since we are
+        # checking None case in the get_translation_contribution_stats_models
+        # function.
+        assert stats_model is not None
```

Comment:
After doing these changes https://github.com/oppia/oppia/pull/15917/files#r963412843  I think you can remove this assert.

Reply:
Done